core/src/main/scala/org/broadinstitute/dsde/rawls/googleProject/GoogleProjectRegistrationService.scala

```diff
@@ -109,4 +112,44 @@ class GoogleProjectRegistrationService(protected val ctx: RawlsRequestContext,
         case None => Future.successful(None)
       }
     } yield finalResult
+
+  def getGoogleProjects(billingProjectName: Option[RawlsBillingProjectName]): Future[Seq[GoogleProjectRegistration]] = {
+    val accessibleGoogleProjectsFuture = samDAO
+      .listResourcesWithActions(SamResourceTypeNames.googleProject, SamGoogleProjectActions.readPolicies, ctx)
+      .map(_.map(resource => GoogleProjectId(resource.getResourceId)).toSet)
+
+    accessibleGoogleProjectsFuture.flatMap { accessibleGoogleProjects =>
+      googleProjectRegRepo.getGoogleProjectRegistrations(accessibleGoogleProjects).map { registrations =>
+        val filteredRegistrations = billingProjectName
+          .map(name => registrations.filter(_.billingProjectId == name))
+          .getOrElse(registrations)
+
+        if (billingProjectName.isDefined && filteredRegistrations.isEmpty) {
+          Seq.empty
+        } else {
+          val registeredProjectIds = filteredRegistrations.map(_.googleProjectId).toSet
+          val unregisteredProjects =
+            accessibleGoogleProjects.diff(registeredProjectIds).map(UnRegisteredGoogleProjectRegistration.apply)
+
+          filteredRegistrations ++ unregisteredProjects
+        }
+      }
+    }
+  }
+
+  def getGoogleProjectById(googleProjectId: GoogleProjectId): Future[Option[GoogleProjectRegistration]] =
+    samDAO
+      .userHasAction(SamResourceTypeNames.googleProject,
+                     googleProjectId.value,
+                     SamGoogleProjectActions.readPolicies,
+                     ctx
+      )
+      .flatMap {
+        case true =>
+          googleProjectRegRepo.getGoogleProjectRegistration(googleProjectId).map {
+            case Some(project) => Some(project)
+            case None          => Some(UnRegisteredGoogleProjectRegistration(googleProjectId))
+          }
+        case false => Future.successful(None)
```

Comment:
if the user does not have permission to read the requested project, or if there is no row in the db for the requested project, throw a `RawlsExceptionWithErrorReport` with an `ErrorReport` specifying status code 404 and the Rawls-standard "does not exist or you do not have permission to use it" message. This will also simplify handling in your ApiService layer; the api layer then doesn't need to handle Some vs. None